modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -1510,7 +1511,8 @@ public Filter[] splitFilter(Filter filter, SimpleFeatureType schema) {
 
                     @Override
                     public Object visit(Function expression, Object notUsed) {
-                        if (expression instanceof JsonPointerFunction) {
+                        if (expression instanceof JsonPointerFunction
+                                || expression instanceof JsonArrayContainsFunction) {
```

Comment:
We need to rely on the function register on the dialect and remove this if clause.